Clarify usage of Verifier and Relying Party (#205)
ietf-wg-scitt/draft-ietf-scitt-architecture: draft-ietf-scitt-architecture.md

: an entity that checks the correctness and consistency of all Transparent Statements issued by a Transparency Service.

> Relying Party:

Author: This PR is still in draft, but I intend to argue here, mailing list, and heaven forbid editor meetings if invited that if client is vague, we should use a precise term and build the necessary semantics around it (while avoiding the implementation-specific details not part of the architecture).

> Relying Party:

Reviewer: ```suggestion
Client:
```

> : a client application for protected Transparency Service resource requests on behalf of the resource owner and with its authorization.

Reviewer 2: ```suggestion
: a relying party depends on or is a Verifier of Signed or Transparent Statements.
```

Reviewer: Personally, I'd like to keep CLient as defined and add Relying Party. I'd decline this suggestion, here.

> The term "Relying Party" does not imply any particular implementation characteristics (e.g., whether the application executes on a server, a desktop, or other devices).

Reviewer: ```suggestion
The term "Client" does not imply any particular implementation characteristics (e.g., whether the application executes on a server, a desktop, or other devices).
```

Transparency Services MUST maintain Registration Policies which govern whether or not a given Signed Statement is eligible for registration.

> Registration Policies MUST be made transparent and available to all Relying Parties of the Transparency Service by registering them as Signed Statements on the Append-only Log.

Author: I am saying we replace the term client with Relying Party like other standards.

Casual usage of client has not helped thus -far, and we should be more specific with a precise term. Additionally, the SCITT architecture is not always a traditional client-server architecture (more complex real-world use cases will function as a multi-layered distributed system), so implying that with use of the word client can further confuse unless we plan to address that.

> Registration Policies MUST be made transparent and available to all Relying Parties of the Transparency Service by registering them as Signed Statements on the Append-only Log.

Reviewer 2: ```suggestion
Registration Policies MUST be made transparent and available to all Relying Parties of the Transparency Service by registering them as Signed Statements on the Append-only Log, and distributing associated Receipts. Distribution of Receipts is out of scope for this document.
```

> Registration Policies MUST be made transparent and available to all Relying Parties of the Transparency Service by registering them as Signed Statements on the Append-only Log.

Reviewer: Yes, by giving a 👍, I was agreeing with you in this context Relying Party was a more narrow clarification 😄

To register a Signed Statement, the Transparency Service performs the following steps:

> 1. **Relying Party authentication:** A Relying Party authenticates with the Transparency Service, to Register Signed Statements.

Author: Part of the original report from a community reviewer and implementer is that "client" as-is is vague and not precise. Casual use of client, in addition to more precision about what this Relying Party does, seems very important to clarifying this person's understanding of the architecture. Same goes for me as well. 😄

> 1. **Relying Party authentication:** A Relying Party authenticates with the Transparency Service, to Register Signed Statements.

Reviewer 2: ```suggestion
1. **Client authentication:** A client application authenticates with the Transparency Service, to Register Signed Statements on behalf of one or more issuers.
```

> 1. **Relying Party authentication:** A Relying Party authenticates with the Transparency Service, to Register Signed Statements.

Reviewer: I believe the added definition of Client solves the gap. As the Issuer isn't the identity that interacts with the Transparency Service, we need a way to describe that entity. A Verifier, Relying Party and Auditor all interact with the Transparency Service through a client. The client is what is authorized to write/read to/from the TS.

Authentication and authorization is implementation-specific, and out of scope of the SCITT Architecture.

> 1. **Issuer Verification:** The Transparency Service MUST perform resolution of the Issuer's identity, which may be different than the Relying Party identity.

Reviewer 2: ```suggestion
1. **Issuer Verification:** The Transparency Service MUST perform resolution of the Issuer's identity.
```

## Transparent Statements {#Receipt}

> The Relying Party (which is not necessarily the Issuer) that registers a Signed Statement and receives a Receipt can produce a Transparent Statement by adding the Receipt to the Unprotected Header of the Signed Statement.

Author: Relying Party will be synonymous with client/Client as we use it today, not Verifier or Issue or other entity.

> The Relying Party (which is not necessarily the Issuer) that registers a Signed Statement and receives a Receipt can produce a Transparent Statement by adding the Receipt to the Unprotected Header of the Signed Statement.

Reviewer 2: ```suggestion
Client applications MAY register Signed Statements on behalf of one or more Issuers.
```

The Transparency Service is trusted with the confidentiality of the Signed Statements presented for Registration.

> Some Transparency Services may publish every Signed Statement in their logs, to facilitate their dissemination and auditing.

Author: I'll defer on that one.

> Others may just return Receipts to Relying Parties that present Singed Statements for Registration, and disclose the Append-only Log only to Auditors trusted with the confidentiality of its contents.

Reviewer 2: ```suggestion
Transparency Services MAY return Receipts to client applications synchronously or asynchronously.
```

The actors depend only on the subset of signing and Receipt schemes they trust.
This enables the gradual transition to stronger algorithms, including e.g. post-quantum signature algorithms.

> ### Transparency Service Relying Parties

Reviewer 2: ```suggestion
### Transparency Service Client Applications
```

> Trust in Relying Parties that submit Signed Statements for Registration is implementation-specific.

Reviewer 2: ```suggestion
Authentication of client applications is out of scope for this document.
```

> An attacker may attempt to register any Signed Statement it has obtained, at any Transparency Service that accepts them, possibly multiple times and out of order.
> This may be mitigated by a Transparency Service that enforces restrictive access control and Registration Policies.

Reviewer 2: ```suggestion
Transparency Services MUST authenticate both client applications and the Issuer of signed statements in order to ensure that implementation specific authentication and authorization policies are enforced.
The specification of authentication and authorization policies is out of scope for this document.
```
